packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -212,7 +212,8 @@ async function generateBase64({ file, args, reporter }) {
   return base64output
 }
 
-const generateCacheKey = ({ file, args }) => `${file.id}${JSON.stringify(args)}`
+const generateCacheKey = ({ file, args, fileArgs }) =>
+  `${file.id}${JSON.stringify(args)}${JSON.stringify(fileArgs) || ''}`
```

Comment:
Using `fileArgs` is the brute force approach that fixed the issue for me. However, I think we might get away with only using `maxWidth` and `maxHeight` instead of using all `fileArgs` to get caching if the same image is used with different qualities, etc.

Reply:
I removed unnecessary properties from the cache key here: 5df0e847c